ARCHITECTURE.md

```diff
@@ -0,0 +1,60 @@
+Project Architecture
+===
+
+## Stack and developer requirements
+
+* This project uses typescript and is built with [NPM](https://www.npmjs.com/get-npm)
+
+### Desktop application requirements
+
+* The desktop application is built with [Tauri](https://tauri.studio/). While this project does not have any significant rust code yet, this may be built out in the future. Running the application as a desktop app requires [Rust](https://www.rust-lang.org/) installed.
+
+## Project design
+
+There are two main concerns separated in this project. Data must be collected from users and that data must be rendered into a PDF file when the user exports their 1040 and attachments.
```

Comment:
In my mind, there are 4 main logical components
* user input pages
* redux storage functionality actions & interfaces
* tax calculations
* PDF form filling

Reply:
Yea important to discuss this part. So far the pdf form filling drives the tax calculations really. For example ach line of the 1040 and other schedules are implemented as separate functions that depend on the output of all the previous lines. 

The only part where we deviate is in the Capital Gains and Qualified Dividends Worksheet which is laborious and designed for someone with a pencil and paper only.

Reply:
I like this four bullet explanation as you've laid it out, I added a drawing about that.

Reply:
> So far the pdf form filling drives the tax calculations really

For sure. The reason I mentioned tax calculations and PDF forms as separate is I think it will be good to keep these areas encapsulated from each other for example if the PDF form fields move around but the calculations stay the same or if we want to add an expected tax preview at each step in or similar in the future.